src/storage/src/source/postgres.rs

```diff
@@ -680,6 +680,14 @@ impl PostgresTaskInfo {
     async fn produce_replication(&mut self) -> Result<(), ReplicationError> {
         use ReplicationError::*;
 
+        // An lsn that is safe to send status updates for. This is primarily derived from
+        // the resumption frontier, as that represents an lsn that is durably recorded
+        // into persist. In the beginning, we can use this initial lsn, which is either:
+        // - From the initial resumption frontier if we are restarting and skipping snapshotting
+        // - The end lsn from the snapshot, which is safe to use because pg keeps
+        //   keeps all updates >= this lsn.
+        let mut committed_lsn: PgLsn = self.lsn;
```

Comment:
@petrosagg should this happen before or after the below correction? I believe its just technically safer to not use the fast-forwarded lsn

Reply:
It should definitely happen before, as you have it here. The way to think about what happens below in timely terms is that we try to advance the LSN **upper** frontier if there are no updates for us. The upper frontier advancing should never cause a change to the commit LSN we send upstream directly.Only indirectly through the normal flow of the dataflow and feedback through resumption frontiers.

However! Even though you've initialized his correctly it is also broken right now (i.e the issue #14961 fixes has regressed). Actually it's broken on `main`, yikes. The reason is the same, we rely on feedback through the resumption frontier operator to send back committed offsets. The code below tries to advance the upper (i.e `self.lsn`) by peeking into the replication slot. **But this is never communicated externally!**. In other words, we don't have a `SourceMessageType::Progress` variant for `get_next_message` to communicate pure progress events that would allow the reclock operator to advance and eventually go the full circle through the resumption frontier operator and into the offset committer.

We should try and fix that, probably in a follow up PR since this didn't break in this PR.

\<insert source-operators-should-be-normal-timely-operators ramble section\>

Reply:
I will write up an issue